File: back/central-reserve/services/horizontalproperty/parking/internal/app/get-parking-types.use-case.go
package app

import (
	"context"

	"central_reserve/services/horizontalproperty/parking/internal/domain"
	"central_reserve/shared/log"
)

// GetParkingTypes obtiene todos los tipos de parqueadero activos
func (uc *parkingUseCase) GetParkingTypes(ctx context.Context) ([]*domain.ParkingType, error) {
	ctx = log.WithFunctionCtx(ctx, "GetParkingTypes")

	types, err := uc.parkingTypeRepo.GetActiveParkingTypes(ctx)
	if err != nil {
		uc.logger.Error().Err(err).Msg("Error obteniendo tipos de parqueadero")
		return nil, err
	}

	return types, nil
}

File: front/rupu-central/src/services/modules/horizontal-properties/parking/application/get-parking-types.use-case.ts
/**
 * UseCase: GetParkingTypesUseCase
 * Obtiene todos los tipos de parqueadero activos
 */

import { ParkingType } from '../domain/entities';
import { IParkingRepository } from '../domain/ports/parking.repository';

export class GetParkingTypesUseCase {
  constructor(private readonly repository: IParkingRepository) {}

  async execute(token: string): Promise<ParkingType[]> {
    return await this.repository.getParkingTypes(token);
  }
}
